safe/utilities/clipper.py

```diff
@@ -676,3 +677,107 @@ def adjust_clip_extent(clip_extent, cell_size, layer_extent):
         adjusted_xmin, adjusted_ymin, adjusted_xmax, adjusted_ymax]
 
     return adjusted_extent
+
+
+def qgis_align_rasters_available():
+    """
+    Determine whether QGIS is running in a version recent enough to contain
+    raster alignment tool (added in 2.12)
+
+    :return: bool
+    """
+    try:
+        from qgis.analysis import QgsAlignRaster
+        return True
+    except ImportError:
+        return False
+
+
+def qgis_align_rasters(hazard_layer, exposure_layer, geo_extent):
+    """
+    Align hazard and exposure raster layers so they fit perfectly and
+    so they can be used for raster algebra. The method uses QGIS
+    raster alignment tool to do the work (which in turn uses GDAL).
+
+    Alignment of layers means that the layers have the same CRS, cell size,
+    grid origin and size. That involves clipping and resampling of rasters.
+
+    From the two layers, the layer with finer resolution (smaller cell size)
+    will be used as the reference for the alignment (i.e. parameters will
+    be set to its CRS, cell size and grid offset).
+
+    Before calling this function make sure to first check whether
+    this functionality is available in QGIS as this requires a recent
+    QGIS version (2.12).
+
+    :param hazard_layer: Hazard layer to be aligned
+    :type exposure_layer: QgsRasterLayer
+
+    :param exposure_layer: Exposure layer to be aligned
+    :type exposure_layer: QgsRasterLayer
+
+    :param geo_extent: Extent in WGS 84 to which raster should be clipped
+    :type geo_extent: list
+
+    :return: clipped hazard and exposure layers
+    :rtype: QgsRasterLayer, QgsRasterLayer
+    """
+
+    hazard_output = unique_filename(suffix='.tif')
+    exposure_output = unique_filename(suffix='.tif')
+
+    exposure_keywords = read_file_keywords(exposure_layer.source())
+
+    # Setup the two raster layers for alignment
+    from qgis.analysis import QgsAlignRaster
+    align = QgsAlignRaster()
+    lst = [QgsAlignRaster.Item(hazard_layer.source(), hazard_output),
+           QgsAlignRaster.Item(exposure_layer.source(), exposure_output)]
+    # TODO: what keywords to consider for rescaling?
+    if exposure_keywords.get('exposure_unit') == 'count':
+        lst[1].rescaleValues = True
+    align.setRasters(lst)
+
+    # Find out which layer has finer grid and use it as the reference.
+    # This will setup destination CRS, cell size and grid origin
+    # TODO: need to respect 'allow_resampling' keyword?
+    if exposure_keywords.get('allow_resampling', 'true').lower() == 'true':
```

Comment:
See this for what I meant https://github.com/AIFDR/inasafe/blob/develop/safe/test/utilities.py#L1181